cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeUtils.java

```diff
@@ -262,4 +346,54 @@ protected void doStop() {
       }
     };
   }
+
+  /**
+   * Uploads the spark event logs through the runtime service.
+   */
+  private static void uploadEventLogs(File eventLogDir, SparkRuntimeContext runtimeContext) throws IOException {
+
+    ProgramRunId programRunId = runtimeContext.getProgramRunId();
+
+    // Find the event file to upload. There should only be one for the current application.
+    File eventFile = Optional.ofNullable(eventLogDir.listFiles())
+      .map(Arrays::stream)
+      .flatMap(Stream::findFirst)
+      .orElse(null);
```

Comment:
Looks like NullPointerException will be thrown if the log directory is empty. Can we expect there to always be exactly one log file?

Reply:
fixed